src/Dapr.Client/InvocationHandler.cs

```diff
@@ -0,0 +1,137 @@
+    /// property will be replaced with a new URI with the authority section replaced by <see cref="DaprEndpoint" />
+    /// and the path portion of the URI rewitten to follow the format of a Dapr service invocation request.
+    /// </para>
+    /// </summary>
+    /// <remarks>
+    /// This message handler does not distinguish between requests destined for Dapr service invocation and general
+    /// HTTP traffic, and will attempt to forward all traffic to the Dapr endpoint. Do not attempt to set
+    /// <see cref="DaprEndpoint" /> to a publicly routable URI, this will result in leaking of traffic and the Dapr
+    /// security token.
+    /// </remarks>
+    public class InvocationHandler : DelegatingHandler
+    {
+        private const string DefaultDaprHttpPort = "3500";
+
+        private static string GetDefaultDaprEndpoint()
+        {
+            var port = Environment.GetEnvironmentVariable("DAPR_HTTP_PORT") ?? DefaultDaprHttpPort;
+            return $"http://127.0.0.1:{port}";
+        }
+
+        private Uri parsedEndpoint;
+        private string? apiToken;
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="InvocationHandler" />.
+        /// </summary>
+        public InvocationHandler()
+        {
+            this.parsedEndpoint = new Uri(GetDefaultDaprEndpoint(), UriKind.Absolute);
+            this.apiToken = Environment.GetEnvironmentVariable("DAPR_API_TOKEN");
+        }
+
+        /// <summary>
+        /// Gets or the sets the URI of the Dapr HTTP endpoint used for service invocation.
+        /// </summary>
+        /// <returns>The URI of the Dapr HTTP endpoint used for service invocation.</returns>
+        public string DaprEndpoint
+        {
+            get
+            {
+                return this.parsedEndpoint.OriginalString;
+            }
+            set
+            {
+                if (value == null)
+                {
+                    throw new ArgumentNullException(nameof(value));
+                }
+
+                // This will throw a reasonable exception if the URI is invalid.
+                this.parsedEndpoint = new Uri(value, UriKind.Absolute);
+            }
+        }
+
+        // Internal for testing
+        internal string? DaprApiToken
+        {
+            get => this.apiToken;
+            set => this.apiToken = value;
+        }
+
+        /// <inheritdoc />
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            var original = request.RequestUri;
+            if (!this.TryRewriteUri(request.RequestUri, out var rewritten))
+            {
+                throw new ArgumentException($"The request URI '{original}' is not a valid Dapr service invocation destination.", nameof(request));
+            }
+
+            try
+            {
+                if (this.apiToken is string)
```

Comment:
Should this be string.IsNullOrEmpty

Reply:
I'll handle this where we read the environment variable. In general it's not a good idea to equate null and empty, but for an env-var it makes sense.